### PR TITLE
Bugfix/make local memory json when it doesnt exist

### PR DESCRIPTION
**Pull request by collijk**
### Background
Fixes a noisy issue where a user selects a local memory backend but does not make an `auto_gpt.json` file.

### Changes
- Make an `auto_gpt.json` file in the workspace if one does not exist.
- Clean up `LocalCache` initialization code to remove a bunch of unreachable conditions.
- Switch test suite to pytest, add more robust testing, mock api calls where reasonable.

### Documentation
No additional docs added, but code is clearer now.

### Test Plan
Overhauled test suite.

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: autogpt/memory/local.py [hunk 1]
```diff
@@ -38,26 +38,16 @@ def __init__(self, cfg) -> None:
...
-                    if not file_content.strip():
-                        file_content = b"{}"
-                        f.write(file_content)
-
-                    loaded = orjson.loads(file_content)
-                    self.data = CacheContent(**loaded)
-            except orjson.JSONDecodeError:
-                print(f"Error: The file '{self.filename}' is not in JSON format.")
-                self.data = CacheContent()
-        else:
```
Comment (collijk) [inline autogpt/memory/local.py:55]: Primary update was to replace this branch of the if/else with a situation where we made the file if it didn't exist.  I also cleaned up a lot of unneccessary logic around initialization in the `if` side of the branch as opening a file in `w+b` mode truncates it, so the file is always empty, which means you always need to add an empty dict and there can never be a json decode error cause there's nothing to decode.

**Review by ntindle — DISMISSED** [review]
LGTM

**Comment by ntindle** [comment]
test fail

**Comment by collijk** [comment]
Oh no, the singleton got me

File: autogpt/llm_utils.py [hunk 2]
```diff
@@ -13,10 +14,54 @@
 from autogpt.types.openai import Message
 
 CFG = Config()
-
 openai.api_key = CFG.openai_api_key
 
 
+def retry_openai_api(
```
Comment (ntindle) [inline autogpt/llm_utils.py]: Would love this to be less openai specific as we work towards including other LLMs

**Review by ntindle — CHANGES_REQUESTED** [review]

File: autogpt/llm_utils.py (hunk 2 above)
Comment (ntindle) [reply autogpt/llm_utils.py]: I’d also keep this goal in mind as you refactor in general

Comment (collijk) [reply autogpt/llm_utils.py]: This commit is a mistake, it's supposed to be in a different PR.  reverting

**Review by ntindle — APPROVED** [review]

**Pull request merged**